### PR TITLE
fix(server): log PR failure at error level

### PR DESCRIPTION
**Pull request by manulpatel**
Close: #6118 

## PR Details

This PR corrects the logging level of PR failure from current `info` level to that of `error`.

### Summary
🐛📝🔧

Fixed a logging bug in the `ActionService` class and improved code consistency. The bug caused errors to be logged with the wrong level, which could affect monitoring and debugging.

> _Sing, O Muse, of the valiant coder who fixed the bug_
> _That plagued the `ActionService` with false alarms and strife_
> _And made the `logError` method speak the truth with `EnumActionLogLevel.Error`_
> _As clear as Zeus's thunderbolts that strike the earth with fire_

### Walkthrough
* Fix the log level and comment of the `logError` method in `action.service.ts` ([link](https://github.com/amplication/amplication/pull/6347/files?diff=unified&w=0#diff-98a64e75d3e418ee905c9980f6f29df46345301e11d6b722ebb44110073f1852L189-R189), [link](https://github.com/amplication/amplication/pull/6347/files?diff=unified&w=0#diff-98a64e75d3e418ee905c9980f6f29df46345301e11d6b722ebb44110073f1852L199-R199))
* Use the `EnumActionLogLevel.Error` constant instead of the `EnumActionLogLevel.Info` constant to log errors ([link](https://github.com/amplication/amplication/pull/6347/files?diff=unified&w=0#diff-98a64e75d3e418ee905c9980f6f29df46345301e11d6b722ebb44110073f1852L199-R199))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/action/action.service.ts [hunk 1]
```diff
@@ -196,6 +196,6 @@ export class ActionService {
     message: string,
     meta: JsonValue = {}
   ): Promise<void> {
-    await this.log(step, EnumActionLogLevel.Info, message, meta);
+    await this.log(step, EnumActionLogLevel.Error, message, meta);
```
Comment (overbit) [inline packages/amplication-server/src/core/action/action.service.ts:199]: @manulpatel thanks for the contribution but this is the correct change unfortunately.
This change will cause all logs to be logged as errors. 
Moreover, the method name clearly state the intention `logInfo` and it should raise some questions about if that is or not the correct place to change the code.

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/action/action.service.ts (hunk 1 above)
Comment (manulpatel) [reply packages/amplication-server/src/core/action/action.service.ts:199]: Very sorry @overbit for the mistake. I found the function responsible to log a PR failure is [here](https://github.com/amplication/amplication/blob/b10f2d21ff6deef833a159563dbd06c7c5922e45/packages/amplication-server/src/core/build/build.service.ts#L642C2-L642C2). Now to log at error level do we need to change input parameters to include error ?

Comment (overbit) [reply packages/amplication-server/src/core/action/action.service.ts:199]: @manulpatel yep, this comment might help https://github.com/amplication/amplication/issues/6118#issuecomment-1596769566

File: packages/amplication-server/src/core/build/build.service.ts [hunk 2]
```diff
@@ -639,7 +639,7 @@ export class BuildService {
     const level = ACTION_LOG_LEVEL[info[LEVEL]];
```
Comment (manulpatel) [inline packages/amplication-server/src/core/build/build.service.ts:639]: May I remove this `level` defined here.

Comment (overbit) [reply packages/amplication-server/src/core/build/build.service.ts:639]: What is calling this `createLog` method? Have you tested this change locally?
I'd suggest to try following the flow in order to fully understand where the change should be. This method doesn't seems to be called and we should probably delete it.

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/build/build.service.ts [hunk 3]
```diff
@@ -449,7 +449,11 @@ export class BuildService {
       step,
       PUSH_TO_GIT_STEP_FAILED_LOG(response.gitProvider)
     );
-    await this.actionService.logInfo(step, response.errorMessage);
+    await this.actionService.log(
+      step,
+      EnumActionLogLevel.Error,
+      response.errorMessage
+    );
```
Comment (overbit) [inline packages/amplication-server/src/core/build/build.service.ts:456]: Very good @manulpatel ! Can you please also update the other logs here to match the correct level (error)?

https://github.com/amplication/amplication/pull/6347/commits/0c6b21075547c24560c84f404d8ffeab440c44a2#diff-0bd5f3f111abe353c918c42195a3ecfc287ced63e6debf18d4d0ddda5a49297aR448

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/core/build/build.service.ts (hunk 3 above)
Comment (overbit) [reply packages/amplication-server/src/core/build/build.service.ts:456]: Please request a re-review once you have done 
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review#further-reading

Comment (manulpatel) [reply packages/amplication-server/src/core/build/build.service.ts:456]: [Here](https://github.com/amplication/amplication/commit/0c6b21075547c24560c84f404d8ffeab440c44a2#diff-0bd5f3f111abe353c918c42195a3ecfc287ced63e6debf18d4d0ddda5a49297aR448) aren't we calling the `logInfo` to log at info level. So is there a need to change this function?

Comment (manulpatel) [reply packages/amplication-server/src/core/build/build.service.ts:456]: Also may I change base branch back to `master` as I need to push some commits to remove the unsed variables to pass lint checks in CI.

**Comment by overbit** [comment]
@manulpatel sorry for the late action. The changes looks good now but we need to rebase/merge master into you branch to avoid unrelated changes. Can you please sync your branch?

**Comment by manulpatel** [comment]
@overbit Thanks for the review! Done rebase to main.

**Review by overbit — APPROVED** [review]

File: packages/amplication-server/src/core/build/build.service.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { Inject, Injectable, forwardRef } from "@nestjs/common";
 import { ConfigService } from "@nestjs/config";
 import { Prisma, PrismaService } from "../../prisma";
 import { LEVEL, MESSAGE, SPLAT } from "triple-beam";
```
Comment (overbit) [inline packages/amplication-server/src/core/build/build.service.ts:4]: @manulpatel you need to remove the unused import to pass the lint checks. Can you please address this?

**Review by overbit — APPROVED** [review]

**Pull request merged**